arm-resources/resources/2016-09-01/swagger/resources.json

```diff
@@ -1853,6 +1853,7 @@
           }
         },
         "debugSetting": {
+          "x-ms-client-flatten": true,
```

Comment:
@jhendrixMSFT - I agree that the flattening extension was previously applied at an incorrect place and this change is trying to apply it correctly. However, this will induce a breaking change in the sdk. This can affect our customers. The sdks have been shipped like this since a long time. I believe customers are used to accessing this as `debugSetting.detailLevel`.

So the right thing would be to
- remove this flattening extension that you have applied correctly
- remove the incorrectly applied flattening extension

Moral of the story: We wont flatten here. This will ensure everything is business as usual. Let me know if my thoughts are not clear.
Note: Have spoken with @vivsriaus and he is in agreement with the recommendation.